lucetc/src/traps.rs

```diff
@@ -1,101 +1,52 @@
 use cranelift_codegen::ir;
 use cranelift_faerie::traps::FaerieTrapManifest;
 
-use byteorder::{LittleEndian, WriteBytesExt};
-use faerie::{Artifact, Decl, Link};
+use faerie::{Artifact, Decl};
 use failure::{Error, ResultExt};
-use std::io::Cursor;
+use lucet_module_data::{TrapManifestRecord, TrapSite};
 
-pub fn write_trap_manifest(manifest: &FaerieTrapManifest, obj: &mut Artifact) -> Result<(), Error> {
-    // declare traptable symbol
-    let manifest_len_sym = "lucet_trap_manifest_len";
-    obj.declare(&manifest_len_sym, Decl::data().global())
-        .context(format!("declaring {}", &manifest_len_sym))?;
+pub fn write_trap_tables(
+    manifest: &FaerieTrapManifest,
+    obj: &mut Artifact,
+) -> Result<Vec<TrapManifestRecord>, Error> {
+    let mut trap_manifest: Vec<TrapManifestRecord> = Vec::with_capacity(manifest.sinks.len());
 
-    let manifest_sym = "lucet_trap_manifest";
-    obj.declare(&manifest_sym, Decl::data().global())
-        .context(format!("declaring {}", &manifest_sym))?;
+    for (i, sink) in manifest.sinks.iter().enumerate() {
+        trap_manifest.push(TrapManifestRecord {
+            table_addr: 0, // This will be fixed up when loaded
```

Comment:
Same hinting question applies here, is there something other than `0` I could place here?